### ISSUE TITLE
Use SLF4J throughout plugin

### ISSUE THREAD

**Issue by pdjohe** [diffplug/spotless#1116]
java.util.Logger is used in many places, but maven builds on SLF4J. The output from the plugin is inconsistent (for example, pom sorting).

```
[INFO] --- spotless-maven-plugin:2.20.1:check (default) @ parent-pom ---
[INFO] Up-to-date checking enabled
[INFO] Index file does not exist. Fallback to an empty index
Feb 03, 2022 1:24:36 PM com.diffplug.spotless.glue.pom.SortPomFormatterFunc$MySortPomLogger info
INFO: Sorting file ...\pom2973277670321689996.xml
Feb 03, 2022 1:24:37 PM com.diffplug.spotless.glue.pom.SortPomFormatterFunc$MySortPomLogger info
INFO: Pom file is already sorted, exiting
[INFO] 
```

All places in the code could be updated to use SLF4J's Logger instead.

Gradle looks like that framework builds on SLF4J too: https://docs.gradle.org/current/javadoc/org/gradle/api/logging/Logger.html

**Comment by nedtwigg** [comment]
Happy to take a PR for this.

**Comment by jamietanna** [comment]
@pdjohe did you want to pick this up? I've had a look at it and if you don't fancy it, I can pick it up :smile:

**Comment by pdjohe** [comment]
@jamietanna : it's all yours! 😄 Happy coding!

**Comment by nedtwigg** [comment]
Published in `plugin-gradle 6.3.0`, not enough new features to justify a maven release yet.
